### PR TITLE
MGMT-11153: Add cluster tags support

### PR DESCRIPTION
**Pull request by danielerez**
Introduced 'tags' support for clusters:
* Added tags property to Cluster entity.
* Support add/update tags in RegisterCluster/UpdateCluster API.
* The tags property in cluster is a comma-separated list of free-text strings.
* It is validated to support tags consist of word characters (a-zA-Z0-9_) and inner whitespaces.
E.g. "tags": "tag1,tag 2,tag3"
* When tags are set for clusters, the feature usage is enabled (for monitoring).

## List all the issues related to this PR

- [x] New Feature 
- [ ] Enhancement 
- [ ] Bug fix
- [ ] Tests
- [ ] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [ ] Automation (CI, tools, etc)
- [x] Cloud
- [ ] Operator Managed Deployments
- [ ] None

## How was this code tested?

- [x] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [ ] Waiting for CI to do a full test run
- [ ] Manual (Elaborate on how it was tested)
- [ ] No tests needed

## Assignees

/cc @avishayt 
/cc @filanov 
/cc @romfreiman
/cc @rccrdpccl  

## Checklist

- [x] Title and description added to both, commit and PR.
- [x] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [x] Reviewers have been listed
- [x] This change does not require a documentation update (docstring, `docs`, README, etc)
- [x] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- Are the title and description (in both PR and commit) meaningful and clear?
- Is there a bug required (and linked) for this change?
- Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md

### CODE REVIEW

File: pkg/validations/validations.go [hunk 1]
```diff
@@ -76,6 +76,9 @@ func AllStrings(vs []string, f func(string) bool) bool {
 }
 
 func ValidateAdditionalNTPSource(commaSeparatedNTPSources string) bool {
+	if commaSeparatedNTPSources == "" {
+		return false
```
Comment (avishayt) [inline pkg/validations/validations.go:80]: Why isn't empty valid?  Isn't it necessary for unsetting the field?

Comment (danielerez) [reply pkg/validations/validations.go:80]: Yeah, but it's actually not relevant for this PR:) I'll remove it.

**Review by avishayt — APPROVED** [review]

**Comment by rccrdpccl** [comment]
Indices are ready now, feel free to unhold whenever

**Comment by danielerez** [comment]
> Indices are ready now, feel free to unhold whenever

Thanks Riccardo!

**Pull request merged**